nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/CommentManagerImpl.java

```diff
@@ -456,4 +471,79 @@ public DocumentModel getThreadForComment(DocumentModel comment) {
         }
         return null;
     }
+
+    @Override
+    public String createComment(CoreSession session, Comment comment) {
+        return CoreInstance.doPrivileged(session, s -> {
+            DocumentModel commentModel = s.createDocumentModel(CommentsConstants.COMMENT_DOC_TYPE);
+            commentModel.setPropertyValue(COMMENT_AUTHOR, comment.getAuthor());
+            commentModel.setPropertyValue(COMMENT_TEXT, comment.getText());
```

Comment:
This looks like the new implementation, you should leverage current one by calling existing method with some transformation.